### PR TITLE
CLI: Improve Storybook packages version management

### PR DESCRIPTION
**Pull request by gaetanmaisse**
## What I did

- ♻️ Refactor script to extract and save versions of all SB packages: 
The previous script was only extracting version for packages under `app` directory. Now, all `@storybook` packages are processed and save with their current version in the monorepo in `versions.json`. This file will be used to find out which version of the packages should be installed when running `sb init`.

- 🆕  Use `versions.json` to get SB packages versions 

- 🗑️ Remove all Storybook packages listed as `devDeps`:
These deps were only there to help to find out which version should be installed when using the CLI. This is now done with a dedicated `versions.json` file so they are not needed in `package.json` anymore. 
Bonus: it removes some `6.0.0-alpha.0` SB packages that were required by `"@storybook/react-native": "6.0.0-alpha.0"`
 
- ➕ Add some dependencies to be able to run `addon-docs` with Yarn 2 out of the box: 
    - `@storybook/addon-docs` is needed as a 1st level dep as we are importing things from `@storybook/addon-docs/blocks` in mdx stories. 
    - `@mdx-js/react`, SB won't deal with mdx file without adding this dep (so this is a temporary work-around). I need to discuss with you @shilman to deeply understand how mdx stories are working.  

 - 🤖 CLI now generates projects with `addon-essential` so it enables `addon-viewport` by default. I added a simple test to check that viewport is properly loaded:
    - Go to Welcome Story
    - Click on Viewport button in the toolbar
    - Select "Small Mobile"
    - Check that Welcome Story is still displayed

## How to test

- CI should be 🟢

### CODE REVIEW

File: lib/cli/package.json [hunk 1]
```diff
@@ -32,10 +32,10 @@
     "versions.json"
   ],
   "scripts": {
-    "prepare": "node ../../scripts/prepare.js && node -r esm ./scripts/generate-app-versions.js",
+    "prepare": "node ../../scripts/prepare.js && node -r esm ./scripts/generate-sb-packages-versions.js",
     "test": "cd test && ./run_tests.sh",
     "test-yarn-2": "cd test && ./run_tests_yarn_2.sh",
-    "postversion": "node -r esm ./scripts/generate-app-versions.js"
+    "postversion": "node -r esm ./scripts/generate-sb-packages-versions.js"
```
Comment (gaetanmaisse) [inline lib/cli/package.json:38]: @shilman can you confirm that with these 2 lines, `versions.json` will be recreated during the release process (after the version bump) and then committed to the repo?

**Comment by gaetanmaisse** [comment]
I discussed a bit with @shilman about mdx and tried to make `addon-docs` work with Yarn 2 without adding `@mdx-js/react` as a dependency but I didn't find a way to do that. So, I suggest merging this PR to fix Yarn 2 compatibility (and merge cli improvements). And give a second look to that when mdx v2 will be released (https://github.com/mdx-js/mdx/issues/1041) and updated in SB.

File: lib/cli/package.json [hunk 2]
```diff
@@ -63,32 +63,6 @@
...
-    "@storybook/channels": "6.0.0-rc.3",
-    "@storybook/ember": "6.0.0-rc.3",
-    "@storybook/html": "6.0.0-rc.3",
-    "@storybook/marionette": "6.0.0-rc.3",
-    "@storybook/marko": "6.0.0-rc.3",
-    "@storybook/mithril": "6.0.0-rc.3",
-    "@storybook/preact": "6.0.0-rc.3",
-    "@storybook/rax": "6.0.0-rc.3",
-    "@storybook/react": "6.0.0-rc.3",
-    "@storybook/react-native": "6.0.0-alpha.0",
```
Comment (gaetanmaisse) [inline lib/cli/package.json:86]: 🦆

**Review by shilman — APPROVED** [review]
Looks great @gaetanmaisse !!! 👏 👏 👏

**Comment by gaetanmaisse** [comment]
@shilman can I merge this PR or should I wait as we are now in RC?

**Pull request merged**